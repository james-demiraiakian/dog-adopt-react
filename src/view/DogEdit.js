import React, { useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useParams } from 'react-router-dom';
import EditDog from '../components/EditDog/EditDog';
import { deleteDog, getDogById, updateDog } from '../services/dogs';

export default function DogEdit() {
  const [name, setName] = useState('');
  const [age, setAge] = useState('');
  const [breed, setBreed] = useState('');
  const [bio, setBio] = useState('');
  const [image, setImage] = useState('');
  const [message, setMessage] = useState('');

  const params = useParams();
  const history = useHistory();

  useEffect(() => {
    const fetchData = async () => {
      const data = await getDogById(params.id);

      setName(data.name);
      setAge(data.age);
      setBreed(data.breed);
      setBio(data.bio);
      setImage(data.image);
    };
    fetchData();
  }, [params.id]);

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await updateDog(params.id, name, bio, age, image, breed);
      setMessage('Dog Updated Sucessfully');

      setTimeout(() => {
        history.push(`/dogs/${params.id}`);
      }, 3000);
    } catch {
      setMessage('Something Went Wrong');
    }
  };

  const handleDelete = async (e) => {
    e.preventDefault();
    try {
      await deleteDog(params.id);
      setMessage('Dog Sucessfully Deleted');

      setTimeout(() => {
        history.push('/');
      }, 3000);
    } catch {
      setMessage('Something went wrong.');
    }
  };

  return (
    <div>
      <h2>Edit Dog</h2>
      <EditDog
        name={name}
        setName={setName}
        bio={bio}
        setBio={setBio}
        age={age}
        setAge={setAge}
        image={image}
        setImage={setImage}
        breed={breed}
        setBreed={setBreed}
        handleSubmit={handleSubmit}
        handleDelete={handleDelete}
      />
      <h1>{message}</h1>
    </div>
  );
}
