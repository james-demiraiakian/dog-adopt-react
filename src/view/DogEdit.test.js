import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import DogEdit from './DogEdit';

test('should render the edit page', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/5/edit']}>
      <Route path="/dogs/:id/edit" component={DogEdit} />
    </MemoryRouter>
  );

  await screen.findByDisplayValue('Caroline');

  expect(container).toMatchSnapshot();
});
