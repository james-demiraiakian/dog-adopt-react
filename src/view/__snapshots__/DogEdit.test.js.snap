// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the edit page 1`] = `
<div>
  <div>
    <h2>
      Edit Dog
    </h2>
    <div>
      <form
        class="dog-form"
      >
        <input
          placeholder="Name"
          type="text"
          value="Caroline"
        />
        <input
          placeholder="Age"
          type="number"
          value="4"
        />
        <input
          placeholder="Breed"
          type="text"
          value="Dingo"
        />
        <input
          placeholder="Bio"
          type="text"
          value="Don't let her eat your baby."
        />
        <input
          placeholder="Image URL"
          type="text"
          value="https://live-production.wcms.abc-cdn.net.au/e81a3a2313e26e1aee1c295d0e508a58?impolicy=wcms_crop_resize&cropH=1688&cropW=3000&xPos=0&yPos=274&width=862&height=485"
        />
        <button>
          Submit
        </button>
        <button>
          Delete
        </button>
      </form>
    </div>
    <h1 />
  </div>
</div>
`;
